src/notification/index.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
+import { hideNotification } from './actions';
+import { Snippet } from '../';
+
+const NOTIFICATION_DURATION = 5000;
+let notificationTimeout;
+
+class Notification extends Component {
+
+  componentDidMount() {
+    this.timer = this.timer.bind(this);
+    this.timer();
+  }
+
+  timer() {
+    if (this.props.timeout && this.props.message) {
+      notificationTimeout = setTimeout(this.props.hideNotification, this.props.timeout);
+    }
```

Comment:
We probably want to clear the timeout somewhere around here too.

In the case where a notification fires, and then 4 seconds later another one fires, the latter would only be displayed for 1 second.

Reply:
I'd also move it outside the condition, to cover even more extreme edge cases where the first notification was se to time out, but the second was not.